File: app/src/main/java/com/example/dagger_hilt/ui/activity/main/adapter/MovieViewHolder.java
package com.example.dagger_hilt.ui.activity.main.adapter;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.dagger_hilt.data.datasource.database.entities.MovieEntity;
import com.example.dagger_hilt.databinding.ItemMovieBinding;
import com.example.dagger_hilt.sys.util.Constants;
import com.squareup.picasso.Picasso;

public class MovieViewHolder extends RecyclerView.ViewHolder {

    private final ItemMovieBinding binding;
    private final MovieInterface movieInterface;

    public MovieViewHolder(@NonNull View itemView, MovieInterface movieInterface) {
        super(itemView);
        binding = ItemMovieBinding.bind(itemView);
        this.movieInterface = movieInterface;
    }

    void bind(MovieEntity movieEntity) {
        binding.setMovie(movieEntity.toMovieEntityKts(movieEntity));
        binding.checkboxLike.setOnCheckedChangeListener((c, b) -> {
            if (movieEntity.getLike() != b) movieInterface.check(movieEntity.getId(), b);
        });
        binding.checkboxLike.setChecked(movieEntity.getLike());
        Picasso.get()
                .load(Constants.BASE_URL_IMAGES + movieEntity.getPosterPath())
//          .placeholder(R.drawable.ic_round_person_24)
//          .error(R.drawable.ic_round_error_24)
                .noFade()
                .into(binding.imagePoster);
    }
}

File: app/src/main/java/com/example/dagger_hilt/ui/activity/main/adapter/MovieViewHolder.kt
package com.example.dagger_hilt.ui.activity.main.adapter

import android.view.View
import android.widget.CompoundButton
import androidx.recyclerview.widget.RecyclerView
import com.example.dagger_hilt.data.datasource.database.entities.MovieEntity
import com.example.dagger_hilt.databinding.ItemMovieBinding
import com.example.dagger_hilt.sys.util.Constants
import com.squareup.picasso.Picasso

class MovieViewHolder(itemView: View, private val movieInterface: (Int, Boolean) -> Unit) :
    RecyclerView.ViewHolder(itemView) {
    private val binding = ItemMovieBinding.bind(itemView)
    fun bind(movieEntity: MovieEntity) {
        binding.movie = movieEntity
        binding.checkboxLike.setOnCheckedChangeListener { _: CompoundButton?, b: Boolean ->
            if (movieEntity.like != b) movieInterface.invoke(
                movieEntity.id,
                b
            )
        }
        binding.checkboxLike.isChecked = movieEntity.like
        Picasso.get()
            .load(Constants.BASE_URL_IMAGES + movieEntity.posterPath)
//          .placeholder(R.drawable.ic_round_person_24)
//          .error(R.drawable.ic_round_error_24)
            .noFade()
            .into(binding.imagePoster)
    }
}